### PR TITLE
split out amp-agnostic parts of bento unit tests tests and creates un…

### PR DESCRIPTION
**Pull request by kvchari**
…it tests for standalone components

* updated tests for following components:
  * base-carousel
  * brightcove
  * facebook
  * inline-gallery
  * iframe
  * instagram
  * bento-lightbox-gallery
  * mathml
  * selector
  * sidebar
  * soundcloud
  * timeago
  * twitter
  * video
  * video-iframe
  * vimeo
  * youtube
  * jwplayer
* removes following tests because they fully overlap with bento versions:
  * test-amp-mathml.js
  * test-amp-video.js
  * test-amp-video-iframe.js because
  * test-amp-vimeo.js
  * test-amp-youtube.js

* doesn't modify test-amp-lightbox-gallery.js, test-amp-sidebar.js because the tests are so different that both are needed

fix bento-selector bug where value changes to options weren't propagated to bento layer

The following components' test are still remaining (i'll handle in a following [PR](https://github.com/ampproject/amphtml/pull/37953))
- social-share
- wordpress-embed
- stream-gallery
- lightbox
- fit-text
- embedly-card

### CODE REVIEW

File: src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js [hunk 1]
```diff
@@ -0,0 +1,352 @@
...
+
+import {defineBentoElement} from '#preact/bento-ce';
+
+import {waitFor} from '#testing/helpers/service';
+
+describes.realWin(
+  'bento-base-carousel:1.0',
+  {
+    amp: false,
+  },
```
Comment (rileyajones) [inline src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js:20]: I know this is copied so don't consider this blocking.

Do we still need `realWin` here? We're trying to get rid of as many of these as possible

File: src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js [hunk 2]
```diff
@@ -0,0 +1,352 @@
...
+    //   const scroller = shadow.querySelector(
+    //     `[class*=${styles.hideScrollbar}]`
+    //   );
+    //   expect(scroller.offsetWidth).to.equal(200);
+
+    //   const prevButton = shadow.querySelector(`[class*=${styles.arrowPrev}]`);
+    //   expect(prevButton.offsetWidth).to.equal(36);
+    //   const nextButton = shadow.querySelector(`[class*=${styles.arrowNext}]`);
+    //   expect(nextButton.offsetWidth).to.equal(36);
+    // });
```
Comment (rileyajones) [inline src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js:350]: Do you have a GitHub issue you can associate this with?
```suggestion
    it.skip('should respect outset-arrows even if controls=never', async () => {
      element.setAttribute('controls', 'never');
      element.setAttribute('outset-arrows', '');
      const userSuppliedChildren = setSlides(3);
      userSuppliedChildren.forEach((child) => element.appendChild(child));
      win.document.body.appendChild(element);

      await element.getApi();

      const shadow = element.shadowRoot;

      // Container is 300px and arrows each take up 50px after padding
      expect(element.offsetWidth).to.equal(300);
      const scroller = shadow.querySelector(
        `[class*=${styles.hideScrollbar}]`
      );
      expect(scroller.offsetWidth).to.equal(200);

      const prevButton = shadow.querySelector(`[class*=${styles.arrowPrev}]`);
      expect(prevButton.offsetWidth).to.equal(36);
      const nextButton = shadow.querySelector(`[class*=${styles.arrowNext}]`);
      expect(nextButton.offsetWidth).to.equal(36);
    });
```

File: src/bento/components/bento-soundcloud/1.0/test/standalone/test-bento-soundcloud.js [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+      // Extract iframe
+      const iframe = element.shadowRoot.querySelector('iframe');
+
+      // Make sure iframe is available
+      expect(iframe).to.not.be.null;
+
+      /** Cannot test as userAssert not available for Preact Component */
+      // expect(iframe.src).to.be.rejectedWith(
+      //   /data-trackid or data-playlistid is required for <bento-soundcloud>/
+      // );
```
Comment (rileyajones) [inline src/bento/components/bento-soundcloud/1.0/test/standalone/test-bento-soundcloud.js:235]: Can this be deleted or else associated with a GitHub issue?

File: src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js (hunk 1 above)
Comment (kvchari) [reply src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js:20]: not sure but we do definitely need the web component tests to run chrome/puppeteer so maybe?

File: src/bento/components/bento-soundcloud/1.0/test/standalone/test-bento-soundcloud.js (hunk 3 above)
Comment (kvchari) [reply src/bento/components/bento-soundcloud/1.0/test/standalone/test-bento-soundcloud.js:235]: this is directly copied from amp test, so i'll delete in both

File: src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js (hunk 2 above)
Comment (kvchari) [reply src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js:350]: looks like it works when i just tested it now, so i'll re-add it

File: src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js (hunk 1 above)
Comment (kvchari) [reply src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js:20]: we can test web comps with [jsdom](https://dev.to/ficusjs/unit-testing-web-components-with-ava-and-jsdom-2ofp) as well, but not sure how common that is

**Comment by kvchari** [comment]
updated with unit test for:
      * embedly-card
      * fit-text
      * lightbox
      * stream-gallery

bento-social-share is weird - the amp and standalone bento components are very different and most of the tests fail anyway so I'm just going to skip that completely. We can add tests in the upcoming fixit sprint

File: extensions/amp-base-carousel/1.0/test/test-amp-base-carousel.js [hunk 4]
```diff
@@ -94,161 +93,6 @@ describes.realWin(
       toggleExperiment(win, 'bento-carousel', false, true);
     });
 
-    it('should render slides and arrows when built', async () => {
```
Comment (dethstrobe) [inline extensions/amp-base-carousel/1.0/test/test-amp-base-carousel.js:97]: Don't we want to leave the old tests in still? So we don't accidently break existing functionality.

**Review by dethstrobe — APPROVED** [review]

File: src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js (hunk 1 above)
Comment (rileyajones) [reply src/bento/components/bento-base-carousel/1.0/test/standalone/test-bento-base-carousel.js:20]: There is an excellent chance that this DOES still need `realWin` but I wanted to draw attention to the the subject.
Both `realWin` and `fakeWin` use jsdom to provide a window object in a different way but there is a potential for tests using `realWin` to effect other tests.

**Review by rileyajones — APPROVED** [review]

**Comment by kvchari** [comment]
@dethstrobe @rileyajones question - i deleted a bunch of amp tests because they overlapped with the corresponding bento-standalone ones. but, it's not guaranteed that just because the bento test works, that the amp test will work (bc they might do extra things in the amp layer, they might interact poorly with amp runtime scripts, etc.). Do you guys think i should delete the tests or keep them?

**Comment by rileyajones** [comment]
@kvchari actually maybe not. Testing AMPs dependance on Bento seems valid and could help increase our confidence during the repo transition.

Plus we can always come back and delete them later.

**Comment by kvchari** [comment]
undid my changes to the amp files

**Pull request merged**